src/main/java/frc/robot/RobotState.java

```diff
@@ -0,0 +1,34 @@
+package frc.robot;
+
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.subsystems.Vision.Vision;
+import frc.robot.subsystems.drivetrain.Drive;
+
+public class RobotState {
+    private static RobotState INSTANCE = null;
+    private Drive swerveDrive;
```

Comment:
You didn't instantiate any of your variables, they are null!

Reply:
Ok, I have instantiated them.